File: example-api/search-obj.php
#!/usr/bin/php
<?php
/*
*       Chyro API Model
*				Copyright (c) 2015:
*						Chyro Conseil <support@chyro.tv>
*						Licensed Under MIT license
*       Install PHP curl
*               http://php.net/manual/en/book.curl.php
*/

/*
        0 - Setup
*/

// Credential
$url = 'HOST'; // Host, like preprod.mycompany.chyro.fr (without http://)
$login = 'LOGIN'; // Login with API read/write
$passwd = 'PASS'; // Password
$output_format = 'json'; // Output format : json or xml


/*
 	1 - Chyro API Class
*/
class ChyroApi {
	private $host;
	private $login;
	private $pass;
	private $cookie = false;
	private $format = 'json';

	public function __construct($host, $login, $pass, $format = 'json') {
		$this->host = $host;
		$this->login = $login;
		$this->pass = $pass;
		$this->format = $format;
	}

	public function auth() {
		$cookie = tempnam ("/tmp", "CURLCOOKIE");
		$ch = curl_init("http://".$this->host."/api/auth/gettoken/format/".$this->format."?user=".$this->login."&password=".$this->pass);
		curl_setopt ($ch, CURLOPT_COOKIEJAR, $cookie);
		curl_setopt ($ch, CURLOPT_RETURNTRANSFER, true);
		$this->cookie = $cookie;
		return curl_exec ($ch);
	}

	public function search($type, $token, $query) {
		if ((!$this->cookie) || (!$token))
			return False;
		$ch = curl_init("http://".$this->host."/api/search/".$type."/format/".$this->format."?query={title=test}&token=".$token);
		curl_setopt ($ch, CURLOPT_COOKIEFILE, $this->cookie);
		curl_setopt ($ch, CURLOPT_RETURNTRANSFER, true);
		return curl_exec ($ch);
	}
}

/*
	2 - Main : example
*/
$api = new ChyroApi($url, $login, $passwd, $output_format);
$token = $api->auth();
echo $api->search('program', $token, '{title=test}');

?>

File: example-api/search-obj.py
#!/usr/bin/python
# -*- coding: utf-8 -*-
#
##       Chyro API Model
##       <support@chyro.tv>
##      Install Python Modules
##              - urllib / urllib2
##              - cookielib
##              - simplejson
import urllib, urllib2, cookielib
import simplejson,os,sys


##
##      Setup
##

host="HOST" # Host, like preprod.mycompany.chyro.fr (without http://)
username="LOGIN" # Login with API read/write
password="PASS" # Password


##
##      Main code
##
class ChyroApi:
	def __init__(self, host, login, passwd, form = 'json'):
		self.host = host
		self.login = login
		self.passwd = passwd
		self.form = form
	
	def initAuth(self):
		self.cj = cookielib.CookieJar()
		self.opener = urllib2.build_opener(urllib2.HTTPCookieProcessor(self.cj))
		login_data = urllib.urlencode({'user' : self.login, 'password' : self.passwd})
		try:
			login = self.opener.open('http://%s/api/auth/gettoken/format/json' % self.host, login_data)
		except urllib2.URLError, e:
			return False
		try:
			login_response = simplejson.load(login)
		except ValueError:
			return False
		return login_response['token']

	def searchApi(self, sheet, token, query):
		try:
			resp = self.opener.open('http://%s/api/search/%s/format/%s?query=%s&token=%s' % (self.host, sheet, self.form, query, token))
		except urllib2.URLError, e:
			return False
		try:
			response = simplejson.load(resp)
		except ValueError:
			return False
		return response


if __name__ == "__main__":
	chyro = ChyroApi(host, username, password)
	token = chyro.initAuth()
	if token:
		print chyro.searchApi('program', token, '{title=test}')




		

